addons/mail/static/src/core/common/message.xml

```diff
@@ -41,7 +41,7 @@
                             </small>
                             <small t-if="isPersistentMessageFromAnotherThread" t-on-click.prevent="openRecord" class="ms-1 text-500">
                                 <t t-if="message.thread.model !== 'discuss.channel'">
-                                    on <a t-att-href="message.resUrl"><t t-esc="message.thread.displayName"/></a>
+                                    on <a t-if="message.thread.displayName" t-att-href="message.resUrl" t-esc="message.thread.displayName"/><em class="pe-1 text-decoration-line-through" t-else="">Deleted document</em>
```

Comment:
Could we improve this? The term 'Document' is quite generic; using something more specific, like 'Deleted Invoice' or 'Deleted Task,' might be clearer. Just a suggestion, though, as I realize this could be a bit challenging.